File: flutter_app/lib/models/notification_item.dart
// import 'package:flutter/material.dart';

/// Notification type; each value is intended to link to a specific screen when tapped.
/// TODO: Use [destinationRoute] when implementing navigation to these screens.
enum NotificationType {
  /// Links to tasks screen (e.g. task list or task detail).
  medication,
  /// Links to messages screen.
  message,
  /// Links to tasks screen (e.g. appointment / calendar).
  appointment,
  /// Links to tasks screen (e.g. health log / reminders).
  healthReminder,
  /// Links to tasks screen (e.g. task list or task detail).
  taskReminder,
}

/// Notification model for app-wide use (e.g. NotificationProvider).
/// Time display is derived from [createdAt]; no stored timeLabel.
class NotificationItem {
  final String id;
  final String title;
  final String summary;
  final NotificationType type;
  final DateTime createdAt;

  bool isRead;

  NotificationItem({
    required this.id,
    required this.title,
    required this.summary,
    required this.type,
    DateTime? createdAt,
    this.isRead = false,
  }) : createdAt = createdAt ?? DateTime.now();

  /// Destination for navigation when the user taps this notification.
  /// TODO: Navigate to the corresponding screen (TasksScreen, MessagesScreen, etc.) when created.
  String get destinationRoute {
    switch (type) {
      case NotificationType.medication:
      case NotificationType.appointment:
      case NotificationType.healthReminder:
      case NotificationType.taskReminder:
        return 'tasks';
      case NotificationType.message:
        return 'messages';
    }
  }

  NotificationItem copyWith({
    String? id,
    String? title,
    String? summary,
    NotificationType? type,
    DateTime? createdAt,
    bool? isRead,
  }) {
    return NotificationItem(
      id: id ?? this.id,
      title: title ?? this.title,
      summary: summary ?? this.summary,
      type: type ?? this.type,
      createdAt: createdAt ?? this.createdAt,
      isRead: isRead ?? this.isRead,
    );
  }
}

File: react_native_app/models/NotificationItem.ts
// =============================================================================
// NOTIFICATION MODEL
// =============================================================================
// Domain model for in-app notifications. Used by NotificationProvider and
// NotificationScreen. destinationRoute maps type to route for tap handling.
// taskId links notification to a specific task in TaskProvider (optional).
// =============================================================================

export enum NotificationType {
  /// Links to tasks screen (e.g. task list or task detail).
  medication = 'medication',
  /// Links to messages screen.
  message = 'message',
  /// Links to tasks screen (e.g. appointment / calendar).
  appointment = 'appointment',
  /// Links to tasks screen (e.g. health log / reminders).
  healthReminder = 'healthReminder',
  /// Links to tasks screen (e.g. task list or task detail).
  generalReminder = 'generalReminder',
}

export interface NotificationItem {
  id: string;
  title: string;
  summary: string;
  type: NotificationType;
  createdAt: Date;
  isRead: boolean;
  taskId?: string; // Optional link to a task in TaskProvider
}

export const getDestinationRoute = (type: NotificationType): string => {
  switch (type) {
    case NotificationType.medication:
    case NotificationType.appointment:
    case NotificationType.healthReminder:
    case NotificationType.generalReminder:
      return '/task-details'; // Navigate to task-details screen
    case NotificationType.message:
      return '/messages'; // Navigate to messages screen
  }
};

export const isTaskRelated = (type: NotificationType): boolean => {
  return (
    type === NotificationType.medication ||
    type === NotificationType.appointment ||
    type === NotificationType.healthReminder ||
    type === NotificationType.generalReminder
  );
};
